### PR TITLE
Adding terraform directory with refactored environment definition

### PR DESCRIPTION
**Pull request by tobafolaranmi-ch**
New features
- fargate
- autoscaling
- global vars
- paramstore versioning

### CODE REVIEW

File: terraform/groups/ecs-service/data.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  name = "${local.name_prefix}-ecs-task-execution-role"
+}
+
+data "aws_lb" "dev_specs_lb" {
+  name = "dev-specs-${var.environment}-lb"
+}
+data "aws_lb_listener" "dev-specs-lb-listener" {
+  load_balancer_arn = data.aws_lb.dev_specs_lb.arn
+  port              = 443
+}
```
Comment (sienkin) [inline terraform/groups/ecs-service/data.tf:43]: Can we please switch this to usding underscores in the block label?

E.g.:
```
data "aws_lb_listener" "dev_specs_lb_listener" {

File: terraform/groups/ecs-service/main.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  # Environmental configuration
+  environment             = var.environment
+  aws_region              = var.aws_region
+  aws_profile             = var.aws_profile
+  vpc_id                  = data.aws_vpc.vpc.id
+  ecs_cluster_id          = data.aws_ecs_cluster.ecs_cluster.id
+  task_execution_role_arn = data.aws_iam_role.ecs_cluster_iam_role.arn
+
+  # Load balancer configuration
+  lb_listener_arn                 = data.aws_lb_listener.dev-specs-lb-listener.arn
```
Comment (sienkin) [inline terraform/groups/ecs-service/main.tf:33]: Please see https://github.com/companieshouse/dapperdox.developer.ch.gov.uk/pull/219/files#r1386201332

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars [hunk 3]
```diff
@@ -0,0 +1,9 @@
+environment = "cidev"
+aws_profile = "development-eu-west-2"
+
+# service configs
+use_set_environment_files = true
+log_level = "trace"
+include_api_filing_public_specs = "1"
+include_pending_public_specs = "1"
+include_private_specs = "1"
```
Comment (sienkin) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:9]: Can these be removed if they're being supplied by the global config?

Comment (tobafolaranmi-ch) [reply terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:9]: Yes they can, I'll remove them for `cidev` only for now

**Review by sienkin — APPROVED** [review]

**Pull request merged**